src/CarbonAware.WebApi/test/unitTests/Controllers/CarbonAwareControllerTests.cs

```diff
@@ -243,4 +244,49 @@ public async Task GetEmissionsDataForLocationsByTime_NoLocations_ThrowsException
         //Assert
         TestHelpers.AssertStatusCode(result, HttpStatusCode.NoContent);
     }
+
+    [Test]
+    public async Task BatchForecastDataAsync_SuccessfulCallReturnsOk()
+    {
+        // Arrange
+        var handler = new Mock<IForecastHandler>();
+        var controller = new CarbonAwareController(this.MockCarbonAwareLogger.Object, CreateEmissionsHandler(new List<EmissionsData>()).Object, handler.Object, CreateLocations(false).Object);
+
+        EmissionsForecastBatchParametersDTO emissionsForecastBatch = new()
+        {
+            SingleLocation = "Sydney",
+            Start = new DateTimeOffset(2022, 3, 1, 0, 0, 0, TimeSpan.Zero),
+            End = new DateTimeOffset(2022, 3, 1, 1, 0, 0, TimeSpan.Zero),
+            Requested = new DateTimeOffset(2022, 3, 1, 0, 0, 0, TimeSpan.Zero),
+            Duration = 60
+        };
+
+        EmissionsData emissionData = new()
+        {
+            Location = emissionsForecastBatch.SingleLocation,
+            Rating = 0.9,
+            Time = DateTime.Now
+        };
+
+        EmissionsForecast emissionsForecast = new()
+        {
+            RequestedAt = new DateTimeOffset(2022, 3, 1, 0, 0, 0, TimeSpan.Zero),
+            GeneratedAt = new DateTimeOffset(2022, 3, 2, 0, 0, 0, TimeSpan.Zero),
+            EmissionsDataPoints = new List<EmissionsData>() { emissionData },
+            OptimalDataPoints = new List<EmissionsData>() { emissionData },
+        };
+
+        handler.Setup(f => f.GetForecastByDateAsync(
+            emissionsForecastBatch.SingleLocation,
+            emissionsForecastBatch.Start,
+            emissionsForecastBatch.End,
+            emissionsForecastBatch.Requested,
+            emissionsForecastBatch.Duration)).ReturnsAsync(emissionsForecast);
+
+        // Act
+        IActionResult result = await controller.BatchForecastDataAsync(new List<EmissionsForecastBatchParametersDTO>() { emissionsForecastBatch });
+
+        //Assert
```

Comment:
Should we also verify that appropriate method in the handler is called once (like we do for other tests)?

Reply:
Good catch! Done 👍